Source/WebCore/platform/mediastream/gstreamer/GStreamerVideoEncoder.cpp

```diff
@@ -359,17 +393,28 @@ static void webkit_webrtc_video_encoder_class_init(WebKitWebrtcVideoEncoderClass
             g_object_set(self->priv->parser.get(), "config-interval", 1, nullptr);
         }, "target-bitrate", setBitrateBitPerSec, "interval-intraframes");
     Encoders::registerEncoder(X264, "x264enc", "h264parse", "video/x-h264",
-        "video/x-h264,alignment=au,stream-format=byte-stream,profile=constrained-baseline",
+        "video/x-h264,alignment=au,stream-format=byte-stream",
         [](WebKitWebrtcVideoEncoder* self) {
             gst_util_set_object_arg(G_OBJECT(self->priv->encoder.get()), "tune", "zerolatency");
             gst_util_set_object_arg(G_OBJECT(self->priv->encoder.get()), "speed-preset", "ultrafast");
-            g_object_set(G_OBJECT(self->priv->encoder.get()), "key-int-max", 15, "threads", 4, nullptr);
+            g_object_set(self->priv->encoder.get(), "key-int-max", 15, "threads", 4, nullptr);
             g_object_set(self->priv->parser.get(), "config-interval", 1, nullptr);
+
+            const auto* structure = gst_caps_get_structure(self->priv->encodedCaps.get(), 0);
+            auto inputCaps = adoptGRef(gst_caps_new_any());
+            if (const char* profileString = gst_structure_get_string(structure, "profile")) {
+                auto profile = makeString(profileString);
```

Comment:
You could create a StringView and save the chars copy.

Reply:
Ah yes, thanks! I should use this more often.